src/plots/plots.js

```diff
@@ -659,6 +659,41 @@ plots.linkSubplots = function(newFullData, newFullLayout, oldFullData, oldFullLa
     }
 };
 
+// This function clears defaults between the first and second pass of
+// supplyDefaults. It exists because otherwise null attributes are
+// supplyDefault'd and inherited as *colors* instead of an actual null
+// attribute which needs to be supplydefaulted by the individual
+// expanded traces.
+plots.clearExpandedTraceDefaultColors = function(expandedTraces) {
+    var colorAttrs, path, trace, i, j;
+
+    // A better check *might* be to explicitly check for a groupby transform
```

Comment:
_would be_ better for sure, but I'm ok with this for now, as this will never get called for transform-less traces.

Maybe we start thinking about adding transform module _categories_ (similar to our trace module [categories](https://github.com/plotly/plotly.js/blob/fbc0296eb031d1c9f25753c3ec1c7e4ea9b693b8/src/traces/scatter/index.js#L37)) to help us out.